pkg/ssh/ssh.go

```diff
@@ -35,71 +36,138 @@ func RegisterFlags(fs *pflag.FlagSet) {
 }
 
 type Interface interface {
-	// Copy is copy local files to remote host
+	// Copy copy local file to remote
 	// scp -r /tmp root@192.168.0.2:/root/tmp => Copy("192.168.0.2","tmp","/root/tmp")
-	// need check md5sum
-	Copy(host, srcFilePath, dstFilePath string) error
-	// CmdAsync is exec command on remote host, and asynchronous return logs
-	CmdAsync(host string, cmd ...string) error
-	// Cmd is exec command on remote host, and return combined standard output and standard error
+	// skip checksum if env DO_NOT_CHECKSUM=true
+	Copy(host, src, dst string) error
+	// CmdAsync exec commands on remote host asynchronously
+	CmdAsync(host string, cmds ...string) error
+	// Cmd exec command on remote host, and return combined standard output and standard error
 	Cmd(host, cmd string) ([]byte, error)
-	//CmdToString is exec command on remote host, and return spilt standard output and standard error
+	// CmdToString exec command on remote host, and return spilt standard output by separator and standard error
 	CmdToString(host, cmd, spilt string) (string, error)
 	Ping(host string) error
 }
 
-type SSH struct {
-	isStdout   bool
-	User       string
-	Password   string
-	PkFile     string
-	PkData     string
-	PkPassword string
-	Timeout    time.Duration
-
-	// private properties
-	localAddress *[]net.Addr
-	clientConfig *ssh.ClientConfig
+var (
+	getAddressesOnce sync.Once
+	localAddresses   *[]net.Addr
+)
+
+func getLocalAddresses() *[]net.Addr {
+	getAddressesOnce.Do(func() {
+		var err error
+		localAddresses, err = iputils.ListLocalHostAddrs()
+		if err != nil {
+			logger.Warn("failed to list local addresses: %v", err)
+		}
+	})
+	return localAddresses
 }
 
-func NewSSHClient(ssh *v2.SSH, isStdout bool) Interface {
-	if ssh.User == "" {
-		ssh.User = v2.DefaultUserRoot
+type Client struct {
+	*ssh.ClientConfig
+	printStdout bool
+}
+
+var _ Interface = &Client{}
+
+func New(opt *Option, opts ...OptionFunc) (*Client, error) {
+	return newFromOptions(opt, opts...)
+}
+
+func newFromOptions(opt *Option, opts ...OptionFunc) (*Client, error) {
+	if opt == nil {
+		opt = NewOption()
 	}
-	address, err := iputils.ListLocalHostAddrs()
-	// todo: return error?
-	if err != nil {
-		logger.Warn("failed to get local address, %v", err)
+	for i := range opts {
+		opts[i](opt)
+	}
+
+	config := &ssh.ClientConfig{
+		User:    opt.User,
+		Timeout: opt.Timeout,
+		Auth:    []ssh.AuthMethod{},
+		HostKeyCallback: func(hostname string, remote net.Addr, key ssh.PublicKey) error {
```

Comment:
using `HostKeyCallback: ssh.InsecureIgnoreHostKey(),` is better ?

Reply:
use it directly will trigger semgrep ci failed with warning, [ref](https://semgrep.dev/r?q=go.lang.security.audit.crypto.insecure_ssh)